From 2fbfdb4fd631a860d9b047cdab3888c217ca1cae Mon Sep 17 00:00:00 2001
From: Yicheng Luo <ethanluoyc@gmail.com>
Date: Tue, 20 Jun 2023 23:22:44 +0100
Subject: [PATCH 2/2] Fix protobuf

---
 python/riegeli/records/BUILD | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/python/riegeli/records/BUILD b/python/riegeli/records/BUILD
index cde6d546..3daa9dad 100644
--- a/python/riegeli/records/BUILD
+++ b/python/riegeli/records/BUILD
@@ -90,5 +90,7 @@ py_library(
 py_proto_library(
     name = "records_metadata_py_pb2",
     srcs = ["records_metadata.proto"],
-    deps = ["@com_google_protobuf//:protobuf_python"],
+    deps = [
+        "@com_google_protobuf//:well_known_types_py_pb2",
+    ],
 )
-- 
2.25.1

